DependencyInjection/Factory/Loader/FileSystemLoaderFactory.php

```diff
@@ -61,4 +63,18 @@ public function addConfiguration(ArrayNodeDefinition $builder)
                 ->end()
             ->end();
     }
+
+    /**
+     * @param string $reference
+     *
+     * @return Reference
+     */
+    private function createLocatorReference($reference)
+    {
+        return new Reference(
+            sprintf('liip_imagine.binary.locator.%s', $reference),
+            ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE,
+            SymfonyFramework::hasDefinitionSharedToggle()
```

Comment:
Good morning @robfrawley, not an easy question. You are right, the code is fine.
I would prefer a feature switch at the moment and remove it in future (maybe 2.0) when at least symfony 2.8 will be supported.